tests/tracer/test_logging.py

```diff
@@ -0,0 +1,284 @@
+import os
+
+
+def test_unrelated_logger_in_debug_with_patch_all(run_python_code_in_subprocess, tmpdir):
+    """
+    When importing the tracer after logging has been configured,
+                with debug mode and enabling a custom log path,
+                the ddtrace logger does not override any custom logs settings and
+                the ddtrace logger continues to have its expected configuration and
+                logs to a custom file path.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    code = """
+import logging
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+import ddtrace
+
+ddtrace.patch_all()
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'RotatingFileHandler'
+assert ddtrace_logger.handlers[0].maxBytes == 15 << 20
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
```

Comment:
did anything get logged? is this why you have the call to `patch_all()`?

Reply:
if so, maybe just add a comment on the call to `patch_all()` to say something like `# calling patch_all() to generate from debug logs` (?)

Reply:
`patch_all` doesn't change the result. I removed it in the latest commit and it had no effect.